src/JsonLd/Serializer/ItemNormalizer.php

```diff
@@ -63,16 +63,18 @@ public function normalize($object, $format = null, array $context = [])
         $resourceClass = $this->resourceClassResolver->getResourceClass($object, $context['resource_class'] ?? null, true);
         $resourceMetadata = $this->resourceMetadataFactory->create($resourceClass);
         $data = $this->addJsonLdContext($this->contextBuilder, $resourceClass, $context);
+        try {
+            $rawData = parent::normalize($object, $format, $context);
+            if (!is_array($rawData)) {
+                return $rawData;
+            }
+            $data['@id'] = $this->iriConverter->getIriFromItem($object);
+            $data['@type'] = $resourceMetadata->getIri() ?: $resourceMetadata->getShortName();
 
-        $rawData = parent::normalize($object, $format, $context);
-        if (!is_array($rawData)) {
-            return $rawData;
+            return $data + $rawData;
+        } catch (\Exception $e) { //  catch EntityNotFoundException without being bound to Doctrine.
+           throw new InvalidArgumentException($e->getMessage(), 404, $e);
```

Comment:
I understand and I agree, but how can we catch the EntityNotFoundException without being bound to doctrine ?